mne/viz/ica.py

```diff
@@ -329,21 +343,52 @@ def plot_ica_properties(ica, inst, picks=None, axes=None, dB=True,
 
     # calculations
     # ------------
+
     if isinstance(inst, BaseRaw):
+        # if reject is set, reject
+        if reject == 'auto':
+            reject = getattr(ica, 'reject_', None)
+        if reject is not None:
+            data = inst.get_data()
+            data, drop_inds = _reject_data_segments(data, ica.reject_,
+                                                    flat=None, decim=None,
+                                                    info=inst.info,
+                                                    tstep=2.0)
+            inst_rejected = RawArray(data, inst.info)
+        else:
+            drop_inds = None
         # break up continuous signal into segments
         from ..epochs import _segment_raw
+        inst_rejected = _segment_raw(inst_rejected, segment_length=2., verbose=False,
```

Comment:
does `inst_rejected` exist when reject is None?

Reply:
if not then this would raise an error, right?

Reply:
BTW - even if it exists at this point you would be segmenting a raw signal that already has missing parts (but I don't remember how `_reject_data_segments` works exactly so I may be wrong), which may lead to having segments with discontinuities. It would be better to first segment the data and only then use `_reject_data_segments`.

Reply:
> _reject_data_segments basically does not work on epochs. It takes the whole data signal (raw.get_data()) and returns data without the parts that were rejected.

Ok, I think we should add some public functions for removing segments/epochs with `reject` dict (not here however). Would it be difficult to adapt `_reject_data_segments` or create a separate `_reject_epochs` that works on epochs and accepts `reject` dict (and not indices like `epochs.drop()` does IIRC)?

Reply:
We could create a helper function ```_reject``` that applies on both epochs and raw and returns dropped_indices and 'cleaned' object. It would not be that difficult to do I believe, but I haven't really worked with epochs yet, so I can't be sure.

Reply:
Yes, but that should be another PR.